cetmix_tower_server/models/cx_tower_server.py

```diff
@@ -1550,70 +1553,60 @@ def _run_python_code(
-            sudo pwd
-            sudo ls -l
+        """
+        Prepare SSH command(s) for execution on a remote server.
+
+        Behavior:
+        - Without sudo: returns the original command string,
+          optionally prefixed with `cd {path} && ` if a path is provided.
+        - With sudo:
+          * If no_split_for_sudo=True and '&&' in command_code:
+            do not split; prepend `sudo -S -p ''` once to the entire command.
+          * Otherwise, split command_code on '&&', strip line-continuation backslashes,
+            and for each segment: prepend `sudo -S -p ''`.
+            For passwordless sudo (mode 'n'), rejoin segments
+            using '&&' into a single string.
 
         Args:
-            command_code (text): initial command
-            path (str, optional): directory where command should be run
-            sudo (str, optional): sudo mode (n, p)
-                n - sudo without password
-                p - sudo with password
-            kwargs (dict):  extra arguments. Use to pass external values.
-                    Following keys are supported by default:
-                        - "log": {values passed to logger}
-                        - "key": {values passed to key parser}
+            command_code (str): original command line, may contain '&&'.
+            path (str, optional): Directory in which to run the commands.
+            sudo (str|bool, optional): Sudo mode:
+                - False or None: run without sudo.
+                - 'p': run with sudo and password (returns a list of commands).
+                - 'n': run with sudo without password (returns a single string).
+            **kwargs: Additional flags:
+                no_split_for_sudo (bool): if True, do not split on '&&'.
 
         Returns:
-            command (list|str): command splitted into separate commands
-                (for sudo with password mode ('p')) or composed command
-                from its parts for sudo without password mode ('n'))
-
+            str or list[str]: Ready-to-execute command(s) for the SSH executor.
         """
-        # Prepare command for sudo if needed
         if sudo:
-            # Add location
             sudo_prefix = "sudo -S -p ''"
+            no_split = kwargs.get("no_split_for_sudo", False)
 
-            # Detect command separator
-            separator = "&&"
-            if separator in command_code:
-                result = (
-                    command_code.replace("\\", "").replace("\n", "").split(separator)
-                )
-
-                # Sudo with password expects a list of commands
-                result = [f"{sudo_prefix} {cmd.strip()}" for cmd in result]
-
-                # Merge back into a single command is sudo is without password
-                if sudo == "n":
-                    result = f" {separator} ".join(result)
-            else:
-                # Single command only
+            if no_split is True:
```

Comment:
The result of this check is the same as when "&&" is not in the command code. You should join these checks in order to to minimise the number of conditional checks.